POEM_047.md

```diff
@@ -0,0 +1,304 @@
+problem has run due to the use of weakrefs. Pickling (or serializing) the object
+can be useful for a variety of reasons such as saving the object or when one is 
+working on the GUI side passing the object between threads or processes.
+Serialization could still be accomplished using a heavier serializer such as 
+Dill, but this add a decent amount of overhead and processes time. Additioanlly, 
+the problem in general adds significant size to the object given that it is only
+being used for local data retrieval. We can imagine a problem where many of
+these obejct are being used and persisting many Problems in memory may not be
+desirable.
+
+## Implentation 2 - Creating of a post_run() function
+A second implementation to allow user interfacing with the data (among other 
+things) could be the creation of a `post_run()` function as part of the core
+OpenMDAO callstack. This would be a method that gets run in the same way as the 
+compute function does when one uses `run_model()` however it gets run after 
+the driver has completed. This would allow the object to do whatever is needed 
+with to the OpenMDAO data while the problem still persists and more generally 
+speaking, do any local cleanup or computations that don't need to be included in
+the `compute` method.
+
+We can imagine a `Parabola.post_run()` function that may look as follows
+
+```python
+    def post_run(self, inputs, outputs):
+        self.y = self.get_val('y', units='mm')
+        self.a = self.get_val('a', units='mm')
+```
+
+And then the `Parabola.u_print_eq()` can simply reference the class attributes
+`a` and `y`.
+
+This method could give users a variety of flexibility when it comes to how they
+choose to interface with the results of an optimization.
+
+## Implementation 3 - No Changes to OpenMDAO
+As an alterante to the above implementations, the class attribute assignment 
+could actually just occur in the regular compute function like follows.
+
+```python
+    def compute(self, inputs, outputs):
+        x = inputs['x']
+        a = inputs['a']
+        y = (x-a)**2 + 4
+        outputs['y'] = y
+        self.x = x
+        self.y = y
+```
+
+These variables would then just be left to the user as they appear in the final 
+iteration. This can be problematic for a variety of reason namely that if
+complex stepping is used, it's possible that these could end up being complex 
+numbers and moreso this really just doesn't seem like the intended use of the 
+`compute()` function.
+
+## Pull Requests
+I have not yet made pull requests, although I have demoed modifications to allow
+each of these implentation to work. I wouldn't be surpised if my modification
+may break other features, so I would like to wait for some community/dev
+feedback before spending too much time perfecting them.
+
+As a brief summary of my initial modidification:
+
+Implentation 1 involves modification to the 
+`System.get_val()` method so that when `from_src` is false, the weakref to the 

```

Comment:
This change seems very reasonable to me.  I would say that if you have a PR with just this change, go ahead and submit it.

Reply:
Thanks for looking this over. PR has been added as #2005